From 65942ca325dffbc8654f8029d83f212b0ca7b5b2 Mon Sep 17 00:00:00 2001
From: djantti <djantti@proton.me>
Date: Sat, 23 Dec 2023 23:08:41 +0200
Subject: [PATCH] udev: add hwdb execution for hidraw subsystem devices

Hwdb call for hidraw subsystem is missing and AV controller devices defined in hwdb.d/70-av-production.hwdb never get the proper permissions for /dev/hidraw*. This patch implements hwdb execution also for hidraw devices.

(cherry picked from commit 43ee987a1f24f390bdee0447022d31ec30f6e5be)

Related: RHEL-5950
---
 rules.d/50-udev-default.rules.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rules.d/50-udev-default.rules.in b/rules.d/50-udev-default.rules.in
index f670b51987..ab29f6a17f 100644
--- a/rules.d/50-udev-default.rules.in
+++ b/rules.d/50-udev-default.rules.in
@@ -14,6 +14,8 @@ SUBSYSTEM=="virtio-ports", KERNEL=="vport*", ATTR{name}=="?*", SYMLINK+="virtio-
 SUBSYSTEM=="rtc", ATTR{hctosys}=="1", SYMLINK+="rtc"
 SUBSYSTEM=="rtc", KERNEL=="rtc0", SYMLINK+="rtc", OPTIONS+="link_priority=-100"
 
+SUBSYSTEM=="hidraw", IMPORT{builtin}="hwdb"
+
 SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", IMPORT{builtin}="usb_id", IMPORT{builtin}="hwdb --subsystem=usb"
 ENV{MODALIAS}!="", IMPORT{builtin}="hwdb --subsystem=$env{SUBSYSTEM}"
 
